### PR TITLE
Pre-warm expression compiler to speed up Flutter Inspector page loading.

### PR DESCRIPTION
**Pull request by annagrin**
Evaluate an expression in widget_inspector library on isolate start to pre-warm the flutter framework module cache in the expression compiler.
     
Flutter inspector relies on evaluations in widget_inspector library, which is a part of the flutter framework module, to produce widget trees, draw the layout explorer, show hover cards etc. Pre-warming the cache while DevTools is still loading helps Flutter Inspector start faster.

Related: https://github.com/dart-lang/sdk/issues/49944 (pre-warming will have speedup effect after this change is propagated to callers of dwds that use expression compiler worker)

Closes: https://github.com/dart-lang/webdev/issues/1768

### CODE REVIEW

File: dwds/debug_extension_mv3/pubspec.yaml [hunk 1]
```diff
@@ -20,6 +20,8 @@ dev_dependencies:
   built_value_generator: ^8.3.0
   build_web_compilers: ^3.0.0
   dwds: ^16.0.0
+  sse: sse: ^4.1.0
```
Comment (annagrin) [inline dwds/debug_extension_mv3/pubspec.yaml:23]: `dart analyze .` in dwds complained about this:)

Comment (elliette) [reply dwds/debug_extension_mv3/pubspec.yaml:23]: It looks like typo? `sse: sse:`

File: dwds/lib/src/services/chrome_proxy_service.dart [hunk 2]
```diff
@@ -203,6 +203,26 @@ class ChromeProxyService implements VmServiceInterface {
...
+    // library, which is a part of the flutter framework module, to
+    // produce widget trees, draw the layout explorer, show hover
+    // cards etc.
+    // Pre-warming the cache while DevTools is still loading helps
+    // Flutter Inspector start faster.
+    final libraryToCache = await inspector.flutterWidgetInspectorLibrary;
+    if (libraryToCache != null) {
+      _logger.finest(
+          'Caching ${libraryToCache.uri} in expression compiler worker');
+      await evaluate(inspector.isolateRef.id!, libraryToCache.id!, 'true');
```
Comment (elliette) [inline dwds/lib/src/services/chrome_proxy_service.dart:222]: Can we check that `inspector.isolateRef.id` and `libraryToCache.id` are not null before passing to `evaluate`?

File: dwds/lib/src/services/chrome_proxy_service.dart [hunk 3]
```diff
@@ -251,6 +271,8 @@ class ChromeProxyService implements VmServiceInterface {
             compiler,
           );
 
+    unawaited(_prewarmExpressionCompilerCache());
```
Comment (elliette) [inline dwds/lib/src/services/chrome_proxy_service.dart:274]: Do we need to check if the `compiler` (above) is `null` first?

File: dwds/lib/src/services/chrome_proxy_service.dart (hunk 2 above)
Comment (annagrin) [reply dwds/lib/src/services/chrome_proxy_service.dart:222]: they are always non-null but I will add a check just to be safe.

Comment (annagrin) [reply dwds/lib/src/services/chrome_proxy_service.dart]: Done!

File: dwds/lib/src/services/chrome_proxy_service.dart (hunk 3 above)
Comment (annagrin) [reply dwds/lib/src/services/chrome_proxy_service.dart:278]: `_prewarmExpressionCompilerCache `will exit without doing anything in that case so not really necessary.

File: dwds/debug_extension_mv3/pubspec.yaml (hunk 1 above)
Comment (annagrin) [reply dwds/debug_extension_mv3/pubspec.yaml]: yes, thanks! fixed.

File: dwds/lib/src/services/chrome_proxy_service.dart (hunk 3 above)
Comment (elliette) [reply dwds/lib/src/services/chrome_proxy_service.dart:278]: OOC, what does the `evaluate` call do in that case?

**Review by elliette — APPROVED** [review]

**Pull request merged**